utils/appliance.py

```diff
@@ -1359,16 +1359,39 @@ def wait_for_ssh(self, timeout=600):
                  delay=5,
                  num_sec=timeout)
 
+    def get_host_address(self, config_name):
+        server = self.get_yaml_config(config_name).get('server', None)
+        if server:
+            return server.get('host', None)
+
+    def wait_for_host_address(self, config_name):
+        """Waits for appliance Host address updated
+
+        Args:
+            config_name: config_name based on appliance version.
+        """
+        try:
+            wait_for(func=self.get_host_address, func_args=[config_name, ],
+                     fail_condition=None,
+                     delay=5,
+                     num_sec=120)
+            return self.get_host_address(config_name)
+        except Exception as e:
+            logger.exception(e)
+            self.log.error('waiting for host address from yaml_config timedout')
+
     @cached_property
     def db_address(self):
         # pulls the db address from the appliance by default, falling back to the appliance
         # ip address (and issuing a warning) if that fails. methods that set up the internal
         # db should set db_address to something else when they do that
         try:
             if self.version >= '5.6':
-                db = self.get_yaml_config("vmdb")['server']['host']
+                db = self.wait_for_host_address('vmdb')
             else:
-                db = self.get_yaml_file('/var/www/miq/vmdb/config/vmdb.yml.db')['server']['host']
+                db = self.wait_for_host_address('/var/www/miq/vmdb/config/vmdb.yml.db')
```

Comment:
mfalesni is right, this is going to break, one used ```get_yaml_file```, the other used ```get_yaml_config```, these details have been lost.